### PR TITLE
Remove the SDK artifacts from the CLI.

### PR DESCRIPTION
**Pull request by eerhardt**
These are now contained in github.com/dotnet/sdk.

Also, tell `dotnet restore` to install nuget.props and nuget.targets files using an environment variable, so the SDK .targets are installed into the user's project.

@piotrpMSFT @livarcocc @brthor

### CODE REVIEW

File: src/dotnet/commands/dotnet-restore/Program.cs [hunk 1]
```diff
@@ -27,8 +27,16 @@ public static int Run(string[] args)
 
             app.OnExecute(() =>
             {
+                const string nugetXProjWriteTargets = "NUGET_XPROJ_WRITE_TARGETS";
+                bool setXProjWriteTargets = Environment.GetEnvironmentVariable(nugetXProjWriteTargets) == null;
+
                 try
                 {
+                    if (setXProjWriteTargets)
+                    {
```
Comment (brthor) [inline src/dotnet/commands/dotnet-restore/Program.cs]: If the environment variable is already set, set it again, and then remove whatever value it had originally..

Why do we need to do this?

Comment (eerhardt) [inline src/dotnet/commands/dotnet-restore/Program.cs]: That way someone else higher up can control what the variable should be. If we blindly overwrite the variable, no one else will get a say in what it should be.  It is generally good to allow your callers to override your defaults. They may know better than you. That being said, it is highly unlikely anyone will ever set this variable outside of us. But if they do, we should respect their wishes.

A thing to note is that this is temporary. Restore is moving to 'MSBuild /t:Restore', which will handle installing .props and .targets files from packages appropriately. This is only needed in the interim.

**Comment by livarcocc** [comment]
No tests to check that the environment variable is being set? Seems like an important business logic of restore.

File: src/dotnet/commands/dotnet-restore/NuGet3.cs [hunk 2]
```diff
@@ -19,11 +16,18 @@ public static int Restore(IEnumerable<string> args)
...
+            var nugetApp = new NuGetForwardingApp(Enumerable.Concat(prefixArgs, args));
 
-            return result;
+            // setting NUGET_XPROJ_WRITE_TARGETS will tell nuget restore to install .props and .targets files
+            // coming from NuGet packages
+            const string nugetXProjWriteTargets = "NUGET_XPROJ_WRITE_TARGETS";
+            bool setXProjWriteTargets = Environment.GetEnvironmentVariable(nugetXProjWriteTargets) == null;
+            if (setXProjWriteTargets)
+            {
+                nugetApp.WithEnvironmentVariable(nugetXProjWriteTargets, "true");
```
Comment (brthor) [inline src/dotnet/commands/dotnet-restore/NuGet3.cs]: I'm confused why we need to do this, won't hte environment variable just pass through appropriately?

Comment (eerhardt) [inline src/dotnet/commands/dotnet-restore/NuGet3.cs]: We only do this when the environment variable isn't set.  So how will it be passed through?

Comment (brthor) [inline src/dotnet/commands/dotnet-restore/NuGet3.cs]: I see, my brain parsed this incorrectly
`bool setXProjWriteTargets = Environment.GetEnvironmentVariable(nugetXProjWriteTargets) == null`

Can we just always set this environment variable and simplify the checking logic then?

Comment (eerhardt) [inline src/dotnet/commands/dotnet-restore/NuGet3.cs]: The idea is that if someone else sets this to 'false', or some other value, we don't want to overwrite what they are setting.

**Comment by eerhardt** [comment]
> No tests to check that the environment variable is being set? Seems like an important business logic of restore.

There isn't a unit test, no.  But there is the test that runs "dotnet new -t msbuild, dotnet restore, dotnet build3" which covers this.  Now that we are using the Core SDK NuGet package that contains the .targets files, this test ensures that "dotnet restore" does the right thing - installs the nuget.props|targets files from NuGet packages.  So that test covers that the environment variable is getting set correctly.

**Comment by eerhardt** [comment]
I've added one more commit, which now hooks the CLI up to the new Core.Sdk NuGet package.  And it cleans up some stuff in "build3" and in our template.  PTAL.

**Comment by eerhardt** [comment]
@livarcocc - all these failures appear to be caused by this change to the build.proj file: https://github.com/dotnet/cli/commit/dc7f1fdf840abcd163795a1a3e92743a2272d683

**Comment by eerhardt** [comment]
@dotnet-bot test this please

Should be unblocked by https://github.com/dotnet/cli/pull/4038

**Pull request merged**